include/net/socket.h

```diff
@@ -672,11 +672,6 @@ static inline int socketpair(int family, int type, int proto, int sv[2])
 	return zsock_socketpair(family, type, proto, sv);
 }
 
-static inline int close(int sock)
```

Comment:
It's expected now to include POSIX header even if there's no intention to use `CONFIG_POSIX_API`?

Reply:
Selected by `CONFIG_NET_SOCKETS_POSIX_NAMES` as well, but yeah.

Could always use `zsock_close()` when we really want to make it clear we're not using POSIX.

Aside from that: looks like POSIX, smells like POSIX, .. :shrug:

Reply:
@rlubos - done!

Reply:
@rlubos @pfalcon 
I almost wonder if it's time to simply remove
`CONFIG_NET_SOCKETS_POSIX_NAMES`.. 🤔 Is there a compelling reason to keep it?

If devs want to use POSIX API calls as a shorthand, then it makes sense to select `CONFIG_POSIX_API`. For networking calls, they really are just simple wrappers around the `zsock_*` versions. For other file-related calls, they're just simple wrappers around the related fdtable calls.

Otherwise use the native Zephyr variants.

Reply:
> Is it expected now to include POSIX header even if there's no intention to use `CONFIG_POSIX_API`?

Technically it doesn't have to be (e.g. `#include <posix/unistd.h>` will still work), but if the intent is to call a standard POSIX function, then it makes sense to include a standard POSIX header and to cater to portability rather than 3rd-party APIs that knowingly conflict with standards.